ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt

```diff
@@ -130,13 +134,37 @@ class CoreVerifier(val node: Node) {
                     responseDom.recursiveChildren("EncryptedAttribute") +
                     responseDom.recursiveChildren("EncryptedID")
         }
+
+        private fun validateEntityIdentifiers(responseDom: Node) {
+            responseDom.recursiveChildren().filter { it.attributeText("Format") == ENTITY }
+                    .forEach { checkEntityIdentifier(it) }
+        }
+
+        private fun checkEntityIdentifier(node: Node) {
+            if (node.attributeNode("NameQualifier") != null ||
+                    node.attributeNode("SPNameQualifier") != null ||
+                    node.attributeNode("SPProvidedID") != null) {
+                throw SAMLComplianceException.create(SAMLCore_8_3_6_a,
+                        message = "No Subject element found.",
+                        node = node)
+            }
+            val nodeValue = node.nodeValue
+            if (nodeValue != null) {
+                if (nodeValue.length > ENTITY_ID_MAX_LEN) {
+                    throw SAMLComplianceException.create(SAMLCore_8_3_6_b,
+                            message = "Length of URI [$nodeValue] is [${nodeValue.length}]",
+                            node = node)
+                }
+            }
+        }
     }
 
     /**
      * Verify response against the Core Spec document
      */
     fun verify() {
         preProcess(node)
+        validateEntityIdentifiers(node)
```

Comment:
✏️ `verifyEntityIdentifiers` instead of `validateEntityIdentifiers`. It might also make more sense to run `verifyCommonDataType` before the new method.